docs/content/preview/troubleshoot/active-session-history.md

```diff
@@ -0,0 +1,350 @@
+| ysql_yb_ash_sampling_interval_ms | Time (in milliseconds) duration between two sampling events (ysql, ycql, yb-tserver). <br>Default: 1000. Changing this flag doesn't require a system restart. |
+| ysql_yb_ash_sample_size | Maximum number of events captured per sampling interval. <br>Default: 500. Changing this flag doesn't require a system restart. |
+
+## Limitations
+
+Note that the following limitations are subject to change as the feature is in [Tech Preview](/preview/releases/versioning/#feature-availability) currently.
+
+- ASH is available per node only. [Aggregations](../../develop/learn/aggregations-ycql/) need to be done by you.
+- ASH is not available for [YB-Master](../../architecture/yb-master/) processes.
+- ASH is available only for foreground activities or queries from customer applications. ASH is not available for most background activities such as backups, restore, remote bootstrap, CDC, tablet splitting. ASH is available for flushes and compactions.
+- ASH does not capture start and end time of wait events.
+
+## SQL views
+
+ASH exposes the following [SQL views](../../explore/ysql-language-features/advanced-features/views/) in each node to analyze and troubleshoot performance issues.
+
+### yb_active_session_history
+
+Get information on wait events for each normalized query, YSQL, or YCQL request.
+
+| Column | Type | Description |
+| :----- | :--- | :---------- |
+| root_request_id | UUID | A 16-byte UUID that is generated per request. Generated for queries at YSQL/YCQL layer. |
+| rpc_request_id | integer | Request ID per RPC. This is not globally unique. However, it is a monotonically increasing number for the lifetime of a YB-TServer. If a YB-TServer restarts, the number starts from 0 again, so it may not be unique across time. However, if there are no restarts, the combination of the server/rpc_request_id is unique. This could be used for advanced use cases later. For example, understanding if the same RPC is being sampled multiple times. |
+| wait_event_component | text | There are three components: YSQL, YCQL, and YB-TServer. |
+| wait_event_class | text | Every wait event will have a class associated with it. |
+| wait_event | text | Provides insight into what the RPC is waiting on. |
+| wait_event_type | text | Type of the wait event such as CPU, WaitOnCondition, Network, Disk IO, and so on. |
+| wait_event_aux | text | Additional information for the wait event. For example, tablet ID for yb-tserver wait events. |
+| top_level_node_id | UUID | 16-byte yb-tserver UUID of the YSQL/YCQL node where the query is being executed. |
+| query_id | bigint | Query ID as seen on the `/statements` endpoint. This can be used to join with [pg_stat_statements](../../explore/query-1-performance/pg-stat-statements/)/ycql_stat_statements. For background activities, query ID will be a known constant (for example, log appender is 1, flush is 2, compaction is 3, consensus is 4). |
```

Comment:
We will introduce more fixed query ids in the future. So it's probably good to have it in a way that suggests `log appender is 1, flush is 2, compaction is 3, consensus is 4` is not an exhaustive list

Also, is it possible to have a link to the heading of ycql_stat_statements?

Reply:
Sure, ill add "and so on" at the end. Also, we do not have docs on "ycql_stat_statements" currently. So I couldn’t link it.